### PR TITLE
feat: add support for 2-4 players

### LINKED ISSUE #31: Add support for 2-4 players
Core changes:
- Update the menu to allow the following configurations:
   1. Screen to select number of players (minimum: 2; maximum: 4)
   2. Screen that shows all the 4 players:
      - Player representation (color)
      - Respective player controls (below tank representation)
      - "Play" button to start the game
- Update Readme with new game inputs

Other points to consider:
- [x] #32
- [x] #33
- [x] #34
- [x] #35

### LINKED ISSUE #33: Assign random color to each player
Note: do not allow the same color for different players.

### PR DESCRIPTION
**Pull request by 201flaviosilva**
## References

- Closes https://github.com/GOOFR-Group/tank-trouble/issues/31
- Closes https://github.com/GOOFR-Group/tank-trouble/issues/33

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -6,22 +6,10 @@ The Tank Trouble
...
-### Player 2
-
-| Key | Description    |
-| --- | -------------- |
-| `w` | move forward   |
-| `s` | move backwards |
-| `a` | rotate left    |
-| `d` | rotate right   |
-| `q` | shoot bullet   |
+| Movement       | Player 1      | Player 2 | Player 3 | Player 4 |
```
Comment (201flaviosilva) [inline README.md]: Achas que ficou fixe assim?

(Fiz assim, só para não estar a repetir 4 vezes a mesma tabela)

File: Scripts/GameManager.gd [hunk 2]
```diff
@@ -7,6 +7,18 @@ var debug :bool setget set_debug_mode
 
 var is_game_over :bool
 
+var COLOR_SETS :PoolColorArray = [
```
Comment (201flaviosilva) [inline Scripts/GameManager.gd]: (Cores ainda não testadas com o sprite real)

File: project.godot [hunk 3]
```diff
@@ -102,6 +102,56 @@ p2_shoot={
 "events": [ Object(InputEventKey,"resource_local_to_scene":false,"resource_name":"","device":0,"alt":false,"shift":false,"control":false,"meta":false,"command":false,"pressed":false,"scancode":81,"physical_scancode":0,"unicode":0,"echo":false,"script":null)
  ]
 }
```
Comment (201flaviosilva) [inline project.godot:104]: Faz-te sentido estes inputs?
(Ainda não tem o joystick)

File: README.md (hunk 1 above)
Comment (Goncalo-Marques) [reply README.md]: It looks a lot better now 💯 

However, I would rename `Kp` to `keypad` (all lower case to keep consistency) for better readability.
What do you think?

Comment (201flaviosilva) [reply README.md]: I just put that because in the Godot, the name is `kp`, but yeah, I will change

File: Scripts/MainScene.gd [hunk 4]
```diff
@@ -0,0 +1,14 @@
...
+onready var Player: PackedScene = preload("res://Prefabs/Player/Tanks/Player.tscn")
+
+const OFFSET = Vector2(70 ,70)
+
+func _ready() -> void:
+	for i in range(GameManager.players.size()):
+		var p = GameManager.players[i]
+		var new_player = Player.instance()
+		
+		new_player.call_deferred("start", p.name, p.color, p.input_code)
```
Comment (201flaviosilva) [inline Scripts/MainScene.gd]: This is probably where you want to generate the player's random position

File: Scripts/GameManager.gd (hunk 2 above)
Comment (201flaviosilva) [reply Scripts/GameManager.gd]: Already tested :)

File: Scripts/GameManager.gd [hunk 5]
```diff
@@ -3,10 +3,38 @@ extends Node
...
+	Color(0,1,0,1), # Green
+	Color(0,0,1,1), # Blue
+	Color(1,1,0,1), # Yellow
+	Color(0,1,1,1), # Cyan
+	Color(1,0,1,1), # Magenta
+	Color(1,1,1,1), # White
+]
+
+## List of players
+var players = []
```
Comment (201flaviosilva) [inline Scripts/GameManager.gd]: I can't find how to set the type of the variable

Comment (Goncalo-Marques) [reply Scripts/GameManager.gd]: At max, we can define that as `var players :Array = []`

**Review by Goncalo-Marques — APPROVED** [review]

**Pull request merged**